compiler/rustc_ty_utils/src/ty.rs

```diff
@@ -295,26 +304,27 @@ impl<'tcx> TypeVisitor<TyCtxt<'tcx>> for ImplTraitInTraitFinder<'_, 'tcx> {
             // the `type_of` of the trait's associated item. If we're using the old lowering
             // strategy, then just reinterpret the associated type like an opaque :^)
             let default_ty = if self.tcx.lower_impl_trait_in_trait_to_assoc_ty() {
-                self
-                    .tcx
-                    .type_of(alias_ty.def_id)
-                    .subst(self.tcx, alias_ty.substs)
+                self.tcx.type_of(shifted_alias_ty.def_id).subst(self.tcx, shifted_alias_ty.substs)
```

Comment:
we really need a `alias_ty.type_of(tcx)` method, this code snippet exists way too often 😆

Reply:
I can add it :D